lib/media/preload_manager.js

```diff
@@ -134,11 +134,8 @@ shaka.media.PreloadManager = class extends shaka.util.FakeEventTarget {
     /** @private {!shaka.util.PublicPromise} */
     this.successPromise_ = new shaka.util.PublicPromise();
 
-    /** @private {!shaka.util.PublicPromise} */
-    this.destroyPromise_ = new shaka.util.PublicPromise();
-
-    // Silence promise failures, in case nobody is catching this.destroyPromise_
-    this.destroyPromise_.catch((error) => {});
+    // Silence promise failures, in case nobody is catching this.successPromise_
+    this.successPromise_.catch((error) => {});
```

Comment:
This is the only part I'm unsure of.  It might mask important errors.  Do we need this for our demo or tests?  Or is it just preventative somehow?

Reply:
It's required for the tests.

I just tried this out to remind myself, and without this we fail to properly boot the tests. It complains about uncaught promise rejections.

Reply:
I think we should fix the tests, then, to catch any preload manager errors.  Even if it's just silenced in the test case, that's better, I think, than silencing it globally in the library.

Reply:
I guess we might be able to change the tests instead? I'm not sure how many tests are broken because apparently an uncaught promise failure in afterAll causes the entire test run to stop, for whatever reason.

Reply:
I'll take a stab at it.

Reply:
I tested with:

```js
./build/test.py --browsers Chrome \
  --no-babel \
  --uncompiled \
  --delay-tests 2
```

`--delay-tests` helped make errors after tests clearer by inserting a delay between each case.  After narrowing it down to Player unit tests, I found out the problem was in start() and waitForFinish() returning two technically separate promises that resolve/reject on the same events/values.  Clarifying that so that start() initiates the action only, makes it so that Player is catching errors in all the right places.